typescript/day3.ts

```diff
@@ -0,0 +1,94 @@
+import fs from 'fs';
+import _ from 'lodash';
+
+export function parseInput(input: string) {
+	const lines = input.split(/\r?\n/).filter((x) => x !== '');
+	const bitLines: string[][] = lines.map((x) => x.split(''));
+	const bits = bitLines.map((row) => row.map((x) => parseInt(x)));
+
+	return bits;
+}
+
+export function calculateGammaFromBits(bits: number[][]) {
+	const gamma = _.unzipWith(bits, function (...bits) {
+		var newValue = _.round(_.sum(bits) / bits.length);
+		return newValue;
+	});
+
+	return gamma;
+}
+
+export function calculateEpsilonFromGamma(gamma: number[]) {
+	const epsilon = gamma.map((x) => 1 - x);
+
+	return epsilon;
+}
+
+export function parseDecimal(bits: number[]) {
+	const bitsToDecimalValue = bits.reduce((previous, current, index) => {
+		const exponent = bits.length - index - 1;
+		const numberToAdd = current * Math.pow(2, exponent);
+		return previous + numberToAdd;
+	}, 0);
+	return bitsToDecimalValue;
+}
```

Comment:
You can replace this entire function with `parseInt(bits.join(""), 2)` 😉

Reply:
I realized that after I finished the solution lol. I thought it would be good to leave it as an example of "there may be a better way to do things." I can change it if you would like.

Reply:
Let's leave it and talk about it on-stream if you're ok with that